framework/src/controller/applicationState.js

```diff
@@ -94,56 +87,39 @@ class System {
 			return false;
 		}
 		const peerHard = parseInt(protocolVersion[0]);
-		const myHard = parseInt(this.headers.protocolVersion[0]);
+		const myHard = parseInt(__private.state.protocolVersion[0]);
 		return myHard === peerHard && peerHard >= 1;
 	}
 
 	/**
-	 * Checks nonce (unique app id) compatibility- compatible when different than given.
-	 *
-	 * @param nonce
-	 * @returns {boolean}
-	 */
-	nonceCompatible(nonce) {
-		if (!nonce) {
-			return false;
-		}
-		return nonce && this.headers.nonce !== nonce;
-	}
-
-	/**
-	 * Updates private broadhash and height values.
+	 * Updates broadhash and height values.
 	 *
 	 * @returns {Promise.<boolean, Error>}
 	 */
-	async update() {
+	async update(blocks) {
```

Comment:
application itself cannot have access to database

Reply:
I don't understand this comment. I've removed the storage from the new applicationState 83ebc756281cd0921587de9c208db7361f4348d1